bndtools.core/src/org/bndtools/core/editors/ImportPackageQuickFixProcessor.java

```diff
@@ -0,0 +1,578 @@
+            final int capacity = 2 * fBuildPath.size() + 1;
+            fBuildPathBundles = new HashSet<>(capacity);
+            fBuildPathVersioned = new HashSet<>(capacity);
+            for (VersionedClause bundleVersion : fBuildPath) {
+                fBuildPathBundles.add(bundleVersion.getName());
+                fBuildPathVersioned.add(bundleVersion);
+            }
+        }
+
+        public List<VersionedClause> getBuildPath() throws CoreException {
+            loadModel();
+            return fBuildPath;
+        }
+
+        public IFile getBndFile() {
+            loadFileInfo();
+            return fBndFile;
+        }
+
+        public void addBundle(String bundle) throws CoreException {
+            addBundle(new VersionedClause(bundle, new Attrs()));
+        }
+
+        public void addBundle(VersionedClause versionedBundle) throws CoreException {
+            loadModel();
+            fBuildPathBundles.add(versionedBundle.getName());
+            if (fBuildPathVersioned.add(versionedBundle)) {
+                fBuildPath.add(versionedBundle);
+            }
+            fBndModel.setBuildPath(fBuildPath);
+            fBndModel.saveChangesTo(fBndDoc);
+            InputStream str;
+            try {
+                str = new ByteArrayInputStream(fBndDoc.get()
+                    .getBytes(fBndFile.getCharset()));
+            } catch (UnsupportedEncodingException e) {
+                throw new CoreException(new Status(IStatus.ERROR, BndtoolsConstants.CORE_PLUGIN_ID, "Invalid encoding for " + fBndFile, e));
+            }
+            fBndFile.setContents(str, IResource.KEEP_HISTORY, fMonitor);
+        }
+
+        public boolean containsBundle(String bundle) throws CoreException {
+            loadModel();
+            return fBuildPathBundles.contains(bundle);
+        }
+
+        // public boolean containsBundle(VersionedClause versionedBundle) throws CoreException {
+        // loadModel();
+        // return fBuildPathVersioned.contains(versionedBundle);
+        // }
+    }
+
+    class AddBundleJob extends WorkspaceJob {
+        private final String fBundle;
+        private final BndBuildPathHandler fHandler;
+
+        public AddBundleJob(IInvocationContext context, String bundle) {
+            super("Adding '" + bundle + "' to Bnd build path");
+            fHandler = getBuildPathHandler(context);
+            fBundle = bundle;
+            IWorkspace workspace = ResourcesPlugin.getWorkspace();
+            IResourceRuleFactory ruleFactory = workspace.getRuleFactory();
+            setRule(ruleFactory.modifyRule(fHandler.getBndFile()));
+        }
+
+        @Override
+        public IStatus runInWorkspace(IProgressMonitor monitor) {
+            try {
+                fHandler.setProgressMonitor(monitor);
+                fHandler.addBundle(fBundle);
+
+                return Status.OK_STATUS;
+            } catch (CoreException e) {
+                return e.getStatus();
```

Comment:
I am thinking you may want to log the exception here?

Reply:
I had assumed that the Eclipse Job framework would take care of properly reporting the error that is returned via the `IStatus` return value, but given that you asked the question I decided to verify this empirically. I have verified that this is indeed the case - not only will Eclipse log the error, but it will pop up a dialog with information about the error. If we log it manually we'll get duplicates in the error log.